src/Gui/Window.cpp

```diff
@@ -53,29 +55,34 @@ WindowParameter::WindowParameter(const char *name)
 
 WindowParameter::~WindowParameter()
 {
+    connParamChanged.disconnect();
 }
 
 /** Sets the group of the window to \a name */
 bool WindowParameter::setGroupName(const char* name)
 {
-  if (_handle.isValid())
-    return false; // cannot change parameter group
+  // cannot change parameter group
+  if (_handle.isValid()) {
+    return false;
+  }
 
   assert(name);
-  std::string prefGroup = name;
-  if (prefGroup.compare(0,15,"User parameter:") == 0 ||
-      prefGroup.compare(0,17,"System parameter:") == 0)
+
+  boost::regex ex("^(User|System)\\s{1}parameter:.*$");
+  if (boost::regex_match(name, ex)) {

```

Comment:
It's probably not really important in this application, but isn't this much slower than the original?

Reply:
I propose to use a regex to avoid an error (and comparing strings can be error prone) due to the number of spaces between "User" ("System") and "parameter". With the regex it is made explicit that there is one and only one space.

Reply:
This function only checks for the parameter managers ("User parameter" or "System parameter") provided by the core system. However, it's possible that (proprietary) extension modules register their own parameter managers.

In order to support  them the function should be implemented as:
```

/** Sets the group of the window to \a name */
bool WindowParameter::setGroupName(const char* name)
{
    // cannot change parameter group
    if (_handle.isValid()) {
        return false;
    }

    assert(name);
    std::string prefGroup = name;
    const auto & list = App::GetApplication().GetParameterSetList();

    auto found = std::find_if(list.begin(), list.end(), [prefGroup](auto item) {
        return boost::starts_with(prefGroup, item.first);
    });

    if (found != list.end()) {
        _handle = App::GetApplication().GetParameterGroupByPath(name);
    }
    else {
        _handle = getDefaultParameter()->GetGroup(name);
    }

    return true;
}
```